azurerm/resource_arm_data_factory_integration_runtime_managed.go

```diff
@@ -0,0 +1,441 @@
+		Name:       &name,
+		Properties: basicIntegrationRuntime,
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, factoryName, name, integrationRuntime, ""); err != nil {
+		return fmt.Errorf("Error creating/updating Data Factory Managed Integration Runtime %q (Resource Group %q, Data Factory %q): %+v", name, resourceGroup, factoryName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, factoryName, name, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Data Factory Managed Integration Runtime %q (Resource Group %q, Data Factory %q): %+v", name, resourceGroup, factoryName, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Data Factory Managed Integration Runtime %q (Resource Group %q, Data Factory %q) ID", name, resourceGroup, factoryName)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmDataFactoryIntegrationRuntimeManagedRead(d, meta)
+}
+
+func resourceArmDataFactoryIntegrationRuntimeManagedRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataFactory.IntegrationRuntimesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	factoryName := id.Path["factories"]
+	name := id.Path["integrationruntimes"]
+
+	resp, err := client.Get(ctx, resourceGroup, factoryName, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Data Factory Managed Integration Runtime %q (Resource Group %q, Data Factory %q): %+v", name, resourceGroup, factoryName, err)
+	}
+
+	d.Set("name", name)
+	d.Set("data_factory_name", factoryName)
+	d.Set("resource_group_name", resourceGroup)
+
+	managedIntegrationRuntime, convertSuccess := resp.Properties.AsManagedIntegrationRuntime()
+	if !convertSuccess {
+		return fmt.Errorf("Error converting integration runtime to managed managed integration runtime %q (Resource Group %q, Data Factory %q)", name, resourceGroup, factoryName)
+	}
+
+	if managedIntegrationRuntime.Description != nil {
+		d.Set("description", managedIntegrationRuntime.Description)
+	}
+
+	if computeProps := managedIntegrationRuntime.ComputeProperties; computeProps != nil {
+		if location := computeProps.Location; location != nil {
+			d.Set("location", location)
+		}
+
+		if nodeSize := computeProps.NodeSize; nodeSize != nil {
+			d.Set("node_size", nodeSize)
+		}
+
+		if numberOfNodes := computeProps.NumberOfNodes; numberOfNodes != nil {
+			d.Set("number_of_nodes", numberOfNodes)
+		}
+
+		if maxParallelExecutionsPerNode := computeProps.MaxParallelExecutionsPerNode; maxParallelExecutionsPerNode != nil {
+			d.Set("max_parallel_executions_per_node", maxParallelExecutionsPerNode)
+		}
+
+		if vnetProps := computeProps.VNetProperties; vnetProps != nil {
+			d.Set("vnet_integration", flattenArmDataFactoryIntegrationRuntimeManagedVnetIntegration(vnetProps))
```

Comment:
We should check for an error here
```suggestion
			if err := d.Set("vnet_integration", flattenArmDataFactoryIntegrationRuntimeManagedVnetIntegration(vnetProps)); err != nil {
			     return fmt.Error("Error setting `vnet_integration`: %+v", err)
			}
```

Reply:
Fixed